lib/install/management/finder.go

```diff
@@ -257,49 +273,49 @@ func (d *Dispatcher) SearchVCHs(computePath string) ([]*vm.VirtualMachine, error
 	return vchs, nil
 }
 
-// searchVCHsFromComputePath searches for VCHs in all child ResourcePools under computePath.
-// The computePath can itself be a ResourcePool, ComputeResource or ClusterComputeResource.
-func (d *Dispatcher) searchVCHsFromComputePath(computePath string) ([]*vm.VirtualMachine, error) {
-	defer trace.End(trace.Begin(computePath, d.op))
-
-	pools, err := d.listResourcePools(path.Join(computePath, "..."))
-
+// search finds the compute resources based on the search path and then retrieves
+// the resouce pools for that compute resource.  Once a list of resource pools is
+// collected search will iterate over the pools and search for VCHs in each pool
+func (d *Dispatcher) search(searchPath string) ([]*vm.VirtualMachine, error) {
+	defer trace.End(trace.Begin(searchPath, d.op))
+	var vchs []*vm.VirtualMachine
+	// find compute resources for the search path
+	resources, err := d.session.Finder.ComputeResourceList(d.op, searchPath)
 	if err != nil {
-		err = errors.Errorf("Failed to search for resource pools for compute path %q: %s", computePath, err)
+		if _, ok := err.(*find.NotFoundError); ok {
+			return nil, nil
```

Comment:
Since this func can now return `nil, nil`, should we update callers to handle this case?

Reply:
The callers iterate over the list via a range, so no risk there...